src/cargo/ops/cargo_compile.rs

```diff
@@ -834,7 +810,28 @@ fn list_rule_targets<'a>(
             }
         }
     }
-    Ok(result)
+    Ok(proposals)
+}
+
+fn list_all_targets<'a>(
```

Comment:
Maybe this could be named slightly differently?  It doesn't list *all* targets, it only returns some of them.  Maybe `filter_all_targets`?

Reply:
Much better, thanks.